diagnostics: race in TestProviderFuncRegistration

go test -race -run=TestProviderFuncRegistration ./diagnostics

```
==================
WARNING: DATA RACE
Write at 0x00c00029a008 by goroutine 15:
  github.com/ledgerwatch/erigon-lib/diagnostics.Context[go.shape.struct { github.com/ledgerwatch/erigon-lib/diagnostics_test.count int }].func1()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:187 +0x96
  github.com/ledgerwatch/erigon-lib/diagnostics_test.TestProviderFuncRegistration()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider_test.go:103 +0x2a3
  testing.tRunner()
      testing/testing.go:1595 +0x261
  testing.(*T).Run.func1()
      testing/testing.go:1648 +0x44

Previous read at 0x00c00029a008 by goroutine 18:
  github.com/ledgerwatch/erigon-lib/diagnostics.Send[go.shape.struct { github.com/ledgerwatch/erigon-lib/diagnostics_test.count int }]()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:161 +0x18e
  github.com/ledgerwatch/erigon-lib/diagnostics_test.TestProviderFuncRegistration.func1()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider_test.go:90 +0xe7
  github.com/ledgerwatch/erigon-lib/diagnostics.ProviderFunc.StartDiagnostics()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:72 +0x3a
  github.com/ledgerwatch/erigon-lib/diagnostics.startProvider()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:136 +0x14d
  github.com/ledgerwatch/erigon-lib/diagnostics.StartProviders.func1()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:124 +0xaa

Goroutine 15 (running) created at:
  testing.(*T).Run()
      testing/testing.go:1648 +0x845
  testing.runTests.func1()
      testing/testing.go:2054 +0x84
  testing.tRunner()
      testing/testing.go:1595 +0x261
  testing.runTests()
      testing/testing.go:2052 +0x8ad
  testing.(*M).Run()
      testing/testing.go:1925 +0xcd7
  main.main()
      _testmain.go:53 +0x2bd

Goroutine 18 (running) created at:
  github.com/ledgerwatch/erigon-lib/diagnostics.StartProviders()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:124 +0x236
  github.com/ledgerwatch/erigon-lib/diagnostics_test.TestProviderFuncRegistration()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider_test.go:99 +0x274
  testing.tRunner()
      testing/testing.go:1595 +0x261
  testing.(*T).Run.func1()
      testing/testing.go:1648 +0x44
==================
==================
WARNING: DATA RACE
Write at 0x00c000292080 by goroutine 15:
  runtime.closechan()
      runtime/chan.go:357 +0x0
  github.com/ledgerwatch/erigon-lib/diagnostics.Context[go.shape.struct { github.com/ledgerwatch/erigon-lib/diagnostics_test.count int }].func1()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:188 +0xc9
  github.com/ledgerwatch/erigon-lib/diagnostics_test.TestProviderFuncRegistration()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider_test.go:103 +0x2a3
  testing.tRunner()
      testing/testing.go:1595 +0x261
  testing.(*T).Run.func1()
      testing/testing.go:1648 +0x44

Previous read at 0x00c000292080 by goroutine 18:
  runtime.chansend()
      runtime/chan.go:160 +0x0
  github.com/ledgerwatch/erigon-lib/diagnostics.Send[go.shape.struct { github.com/ledgerwatch/erigon-lib/diagnostics_test.count int }]()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:161 +0x1aa
  github.com/ledgerwatch/erigon-lib/diagnostics_test.TestProviderFuncRegistration.func1()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider_test.go:90 +0xe7
  github.com/ledgerwatch/erigon-lib/diagnostics.ProviderFunc.StartDiagnostics()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:72 +0x3a
  github.com/ledgerwatch/erigon-lib/diagnostics.startProvider()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:136 +0x14d
  github.com/ledgerwatch/erigon-lib/diagnostics.StartProviders.func1()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:124 +0xaa

Goroutine 15 (running) created at:
  testing.(*T).Run()
      testing/testing.go:1648 +0x845
  testing.runTests.func1()
      testing/testing.go:2054 +0x84
  testing.tRunner()
      testing/testing.go:1595 +0x261
  testing.runTests()
      testing/testing.go:2052 +0x8ad
  testing.(*M).Run()
      testing/testing.go:1925 +0xcd7
  main.main()
      _testmain.go:53 +0x2bd

Goroutine 18 (running) created at:
  github.com/ledgerwatch/erigon-lib/diagnostics.StartProviders()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider.go:124 +0x236
  github.com/ledgerwatch/erigon-lib/diagnostics_test.TestProviderFuncRegistration()
      github.com/ledgerwatch/erigon-lib/diagnostics/provider_test.go:99 +0x274
  testing.tRunner()
      testing/testing.go:1595 +0x261
  testing.(*T).Run.func1()
      testing/testing.go:1648 +0x44
==================
--- FAIL: TestProviderFuncRegistration (4.00s)
    testing.go:1465: race detected during execution of test
FAIL
```

--- a/erigon-lib/diagnostics/provider.go
+++ b/erigon-lib/diagnostics/provider.go
@@ -80,30 +80,6 @@ type registry struct {
 var providers = map[Type]*registry{}
 var providerMutex sync.RWMutex
 
-func RegisterProvider(provider Provider, infoType Type, logger log.Logger) {
-	providerMutex.Lock()
-	defer providerMutex.Unlock()
-
-	reg := providers[infoType]
-
-	if reg != nil {
-		for _, p := range reg.providers {
-			if p == provider {
-				return
-			}
-		}
-	} else {
-		reg = &registry{}
-		providers[infoType] = reg
-	}
-
-	reg.providers = append(reg.providers, provider)
-
-	if reg.context != nil {
-		go startProvider(reg.context, infoType, provider, logger)
-	}
-}
-
 func StartProviders(ctx context.Context, infoType Type, logger log.Logger) {
 	providerMutex.Lock()
 

--- a/erigon-lib/diagnostics/provider_test.go
+++ b/erigon-lib/diagnostics/provider_test.go
@@ -17,10 +17,7 @@ func (ti testInfo) Type() diagnostics.Type {
 	return diagnostics.TypeOf(ti)
 }
 
-type testProvider struct {
-}
-
-func (t *testProvider) StartDiagnostics(ctx context.Context) error {
+func StartDiagnostics(ctx context.Context) error {
 	timer := time.NewTicker(1 * time.Second)
 	defer timer.Stop()
 
@@ -39,64 +36,11 @@ func (t *testProvider) StartDiagnostics(ctx context.Context) error {
 
 func TestProviderRegistration(t *testing.T) {
 
-	// diagnostics provider
-	provider := &testProvider{}
-	diagnostics.RegisterProvider(provider, diagnostics.TypeOf(testInfo{}), log.Root())
-
 	// diagnostics receiver
 	ctx, ch, cancel := diagnostics.Context[testInfo](context.Background(), 1)
 	diagnostics.StartProviders(ctx, diagnostics.TypeOf(testInfo{}), log.Root())
 
-	for info := range ch {
-		if info.count == 3 {
-			cancel()
-		}
-	}
-}
-
-func TestDelayedProviderRegistration(t *testing.T) {
-
-	time.AfterFunc(1*time.Second, func() {
-		// diagnostics provider
-		provider := &testProvider{}
-		diagnostics.RegisterProvider(provider, diagnostics.TypeOf(testInfo{}), log.Root())
-	})
-
-	// diagnostics receiver
-	ctx, ch, cancel := diagnostics.Context[testInfo](context.Background(), 1)
-	diagnostics.StartProviders(ctx, diagnostics.TypeOf(testInfo{}), log.Root())
-
-	for info := range ch {
-		if info.count == 3 {
-			cancel()
-		}
-	}
-}
-
-func TestProviderFuncRegistration(t *testing.T) {
-
-	// diagnostics provider
-	diagnostics.RegisterProvider(diagnostics.ProviderFunc(func(ctx context.Context) error {
-		timer := time.NewTicker(1 * time.Second)
-		defer timer.Stop()
-
-		var count int
-
-		for {
-			select {
-			case <-ctx.Done():
-				return nil
-			case <-timer.C:
-				diagnostics.Send(testInfo{count})
-				count++
-			}
-		}
-	}), diagnostics.TypeOf(testInfo{}), log.Root())
-
-	// diagnostics receiver
-	ctx, ch, cancel := diagnostics.Context[testInfo](context.Background(), 1)
-
-	diagnostics.StartProviders(ctx, diagnostics.TypeOf(testInfo{}), log.Root())
+	go StartDiagnostics(ctx)
 
 	for info := range ch {
 		if info.count == 3 {

--- a/turbo/snapshotsync/snapshotsync.go
+++ b/turbo/snapshotsync/snapshotsync.go
@@ -351,10 +351,6 @@ func WaitForDownloader(ctx context.Context, logPrefix string, headerchain, blobs
 	logEvery := time.NewTicker(logInterval)
 	defer logEvery.Stop()
 
-	/*diagnostics.RegisterProvider(diagnostics.ProviderFunc(func(ctx context.Context) error {
-		return nil
-	}), diagnostics.TypeOf(diagnostics.DownloadStatistics{}), log.Root())*/
-
 	// Check once without delay, for faster erigon re-start
 	stats, err := snapshotDownloader.Stats(ctx, &proto_downloader.StatsRequest{})
 	if err != nil {

